rubin_scheduler/scheduler/detailers/detailer.py

```diff
@@ -104,6 +107,76 @@ def __call__(self, observation_list, conditions):
         return ObservationArray()
 
 
+class NSnapsFromExptimeDetailer(BaseDetailer):
+    """If exposure time is low, force number of snaps to be 1.
+
+    Parameters
+    ----------
+    min_exptime : `float`
+        The minimum exposure time. Set anything below
+        to have nexp=1. Default 29 (seconds).
+    """
+
+    def __init__(self, min_exptime=29.0):
+        self.min_exptime = min_exptime
+
+    def __call__(self, observation_array, conditions):
+        indx = np.where(observation_array["exptime"] < self.min_exptime)
+        observation_array["nexp"][indx] = 1
+        return observation_array
+
+
+class SplitLongExp(BaseDetailer):
+    """Split long exposure times into multiple nexp
+
+    Parameters
+    ----------
+    split_long_max : `float`
+        Maximum exposure time. Default 30 (seconds).
+    split_long_div : `float`
+        Used to determine how many snaps to break
+        visit into. Probably should be 2x split_long_max.
+        Default 60 (seconds).
+    """
+
+    def __init__(
+        self,
+        split_long_max=30.0,
+        split_long_div=60.0,
+    ):
+        self.split_long_max = split_long_max
+        self.split_long_div = split_long_div
+
+    def __call__(self, observation_array, conditions):
+        indx = np.where(observation_array["exptime"] > self.split_long_max)
+        observation_array["nexp"][indx] = np.ceil(
+            observation_array["exptime"][indx] / self.split_long_div
+        ).astype(int)
```

Comment:
I think you're going to walk me through what you were aiming for here. 
If the max acceptable exposure time is 30s (split_long_max) and you have a 90s requested exposure time, wouldn't you want it split into 3 separate 30s exposures? with split_long_div and the math here, you're going to get 2 snaps, not 3. 

It may also be worth noting: when we set nexp > 1, these really do turn into snaps downstream, not separate visits. I'm not entirely certain that DM are handling these as we'd expect -- certainly the acquisition of the visits is different than if they were separate visits.

Reply:
Ok - I don't know if DM will have other issues (got a question in #dm-algorithms-pipelines) but I do see that the snaps we took by accident on 20251114 (71-84) ended up a bit messed up in the ConsDb. 

I think, given that nothing truly supports snaps in the science records in consdb, that somewhere here we need to make nexp always be 1, and if we need multiples, they turn into actual multiple visits.

Reply:
ah, this shouldn't be in a detailer anyway because the `scheduler_note` index won't get updated properly.

Reply:
Ah. Hmm. 
So we should just make people set the number of visits and exposure times in the configuration for the scripted survey, and drop all of the various "split exposure time" aspects from the survey plus the detailers?